scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -1103,26 +1131,68 @@ class Router(formatOps: FormatOps) {
                 ignore = x => excludeRanges.exists(_.contains(x.left.start))
               )
             }
-            val jsNative = isJsNative(right)
-            val noNewline = jsNative
-            val spacePolicy: Policy = rhs match {
-              case _: Term.If => twoBranches
-              case _: Term.ForYield => twoBranches
-              case _: Term.Try | _: Term.TryWithHandler
-                  if style.activeForEdition_2019_11 && !noNewline =>
-                null // we force newlines in try/catch/finally
-              case _ => NoPolicy
+            val okNewline = !isJsNative(right)
+            val spacePolicy: Policy = style.newlines.source match {
```

Comment:
It would be good to extract `newlines.source` machinery to another function. Main pattern matching block in `Router` is giant enough :c

Reply:
will do, makes sense. will update the PR shortly.

Reply:
done